be/src/runtime/data_stream_recvr.cc

```diff
@@ -460,8 +460,11 @@ Status DataStreamRecvr::SenderQueue::add_chunks(const PTransmitChunkParams& requ
     ChunkQueue chunks;
     size_t total_chunk_bytes = 0;
     faststring uncompressed_buffer;
+    bool prev_is_pipeline_level_shuffle = _is_pipeline_level_shuffle;
     _is_pipeline_level_shuffle =
             _recvr->_is_pipeline && request.has_is_pipeline_level_shuffle() && request.is_pipeline_level_shuffle();
+    // If _is_pipeline_level_shuffle is already set to true, then it will never go back to false
+    DCHECK(!prev_is_pipeline_level_shuffle || _is_pipeline_level_shuffle);
```

Comment:
could use xor operate.

Reply:
false -> true is allowed
true -> false is not allowed